apis/workloads/v1alpha1/instanceset_types.go

```diff
@@ -75,6 +75,66 @@ type SchedulingPolicy struct {
 	TopologySpreadConstraints []corev1.TopologySpreadConstraint `json:"topologySpreadConstraints,omitempty"`
 }
 
+// InstanceSetUpdateStrategyType is a string enumeration type that enumerates
+// all possible update strategies for the InstanceSet controller.
+// +enum
+type InstanceSetUpdateStrategyType string
+
+const (
+	// RollingUpdateInstanceSetStrategyType indicates that update will be
+	// applied to all Pods in the InstanceSet with respect to the InstanceSet
+	// ordering constraints. When a scale operation is performed with this
+	// strategy, new Pods will be created from the specification version indicated
+	// by the InstanceSet's updateRevision.
+	RollingUpdateInstanceSetStrategyType InstanceSetUpdateStrategyType = "RollingUpdate"
+	// OnDeleteInstanceSetStrategyType triggers the legacy behavior. Version
+	// tracking and ordered rolling restarts are disabled. Pods are recreated
+	// when they are manually deleted. When a scale
+	// operation is performed with this strategy, specification version indicated
+	// by the InstanceSet's currentRevision.
+	OnDeleteInstanceSetStrategyType InstanceSetUpdateStrategyType = "OnDelete"
+)
+
+// InstanceUpdateStrategy indicates the strategy that the InstanceSet
+// controller will use to perform updates. It includes any additional parameters
+// necessary to perform the update for the indicated strategy.
+type InstanceUpdateStrategy struct {
+	// Type indicates the type of the InstanceUpdateStrategy.
+	// Default is RollingUpdate.
+	// +optional
+	Type InstanceSetUpdateStrategyType `json:"type,omitempty"`
+	// RollingUpdate is used to communicate parameters when Type is RollingUpdateStatefulSetStrategyType.
+	// +optional
+	RollingUpdate *RollingUpdateStrategy `json:"rollingUpdate,omitempty"`
+	// Members(Pods) update strategy.
+	//
+	// - serial: update Members one by one that guarantee minimum component unavailable time.
+	// - bestEffortParallel: update Members in parallel that guarantee minimum component un-writable time.
+	// - parallel: force parallel
+	//
+	// +kubebuilder:validation:Enum={Serial,BestEffortParallel,Parallel}
+	// +optional
+	MemberUpdateStrategy *MemberUpdateStrategy `json:"memberUpdateStrategy,omitempty"`
+}
+
+// RollingUpdateStrategy is used to communicate parameter for RollingUpdateInstanceSetStrategyType.
+type RollingUpdateStrategy struct {
+	// Partition indicates the ordinal at which the InstanceSet should be partitioned
+	// for updates. During a rolling update, all pods from ordinal Replicas-1 to
+	// Partition are updated. All pods from ordinal Partition-1 to 0 remain untouched.
+	// This is helpful in being able to do a canary based deployment. The default value is 0.
+	// +optional
+	Partition *int32 `json:"partition,omitempty"`
```

Comment:
Do we need `partition`?  @free6om 

After introducing instance templates, the naming of Pods managed by an InstanceSet follows this pattern: "comp-0, comp-1, comp-tpl0-0, comp-tpl1-0, comp-tpl1-1"

Unlike before, the Pods no longer have a linear ordinal numbering scheme. This makes specifying a `partition` much more challenging.

Reply:
Yes, I feel the same way. Currently, the partition feature allows Pods to be updated in a rolling update based on dictionary order from largest to smallest, but it seems there is no way to perform a rolling update on a specific Template at the moment.

Reply:
How we do a `RollingUpdate` then?